clang/test/Driver/sycl-offload.cpp

```diff
@@ -178,3 +178,10 @@
 // RUN: %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64_gen -fno-sycl-remove-unused-external-funcs %s 2>&1 | FileCheck -check-prefix=CHECK_SYCL_POST_LINK_OPT_NO_PASS %s
 // CHECK_SYCL_POST_LINK_OPT_NO_PASS-NOT: sycl-post-link{{.*}}emit-only-kernels-as-entry-points
 
+/// Check for correct handling of -fsycl-fp64-conv-emu option for different targets
+// RUN: %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=spir64 -fsycl-fp64-conv-emu %s 2>&1 | FileCheck -check-prefix=CHECK_WARNING %s
+// CHECK_WARNING: warning: Invalid option. '-fsycl-fp64-conv-emu' option is supported only for AOT compilation of Intel GPUs [-Winvalid-command-line-argument]
+// RUN: %clang -### -target x86_64-unknown-linux-gnu -fsycl -fsycl-targets=intel_gpu_pvc -fsycl-fp64-conv-emu %s 2>&1 | FileCheck -check-prefix=CHECK_FSYCL_FP64_CONV_EMU %s
```

Comment:
Since `-fsycl-fp64-conv-emu` is visible in `ClangOption, CLOption, DXCOption, CC1Option` modes, can a test case be added for each of these modes?

Reply:
I will add a test for CLOption and CC1Option. Thanks

Reply:
Added testing for CLOption. CC1Option is internally tested in E2E tests. Thanks